### ISSUE TITLE
Unable to deconstruct custom vendors

### ISSUE THREAD

**Issue by nickup9** [tgstation/tgstation#54992]
## Testmerges: 

- https://github.com/tgstation/tgstation/pull/54483
- https://github.com/tgstation/tgstation/pull/54702

## Reproduction:
TLDR: Try multiple ways to deconstruct a custom vending machine, fail multiple times.
- As chef, build a custom vendor, link it to ID
- Try to deconstruct it to change the style after you've built it (not realizing that changing the vendor's style changes the entire vendor, not just the style of the custom vendor)
- Use a screwdriver, then crowbar as per wiki
- Fail
- Try to cut all wires, get shocked, ask assistant and cyborg for help cut the last one, both of them get crushed for the trouble
- Ask an engineer with insuls to cut the wires, try to use the crowbar again
- Fail
- CE comes by and tries to help, they re-mend all the wires and we try to crowbar it again (we all tried different intents during this, I think, with the crowbar)
- Fail
- Collectively scratch heads
- [THIS IS WHAT V2 WAS FOR](https://pbs.twimg.com/media/DzjQXetWkAAf7ET.jpg)
- Admemes fire a missile at the vendor, blowing a leg off a engi
- Vendor gets trashed and machine board pops out, ending the whole debacle.

**Comment by Tlaltecuhtli** [comment]
its intended so someone cant just get decon to get your shit for free

**Comment by TheBonded** [comment]
> its intended so someone cant just get decon to get your shit for free

then there should be a lock function that can only be unlocked with the linked ID or account.
